src/vs/editor/contrib/codeAction/browser/codeAction.ts

```diff
@@ -57,7 +57,7 @@ class ManagedCodeActionSet extends Disposable implements CodeActionSet {
 		}
 	}
 
-	public readonly validActions: readonly CodeActionItem[];
+	public readonly validActions: CodeActionItem[];
```

Comment:
Instead of making this mutable, can you construct a new instance of `CodeActionSet` with the new actions?